src/main/java/com/kodilla/ecommercee/domain/Cart.java

```diff
@@ -26,4 +26,8 @@
     @ManyToOne
     @JoinColumn(name = "USER_ID")
     private User user;
+
+    public Cart(Long id) {
+        this.id = id;
+    }
```

Comment:
Wydaje mi się, żę nie trzeba takiego konstruktora. Cart tworzysz konstruktorem bezparametrowym. Baza danych chyba automatycznie nadaje Ci ID

Reply:
w kontrolerze tworząc nowy cart używasz metody: public void createCart(CartDto cartDto)
otrzymujesz obiekt CartDto który trzeba zmapować na Cart, uznałem że ten konstruktor będzie potrzebny żeby Cart nie był nullem, właśnie takie rzeczy musimy przemyśleć

Reply:
Tez uwazam, ze nie trzeba dodawac tego konstruktora.

Reply:
To może zostawimy te konstruktory, które mamy a dodatkowo nad klasą dodamy @AllArgsKonstruktor do mapowania encji?